nucleus/dataset.py

```diff
@@ -184,6 +187,17 @@ def is_scene(self) -> bool:
         self._is_scene = response
         return self._is_scene  # type: ignore
 
+    @property
+    def use_privacy_mode(self) -> bool:
+        """Whether or not the dataset was created for privacy mode."""
+        if self._use_privacy_mode is not None:
+            return self._use_privacy_mode
+        response = self._client.make_request(
+            {}, f"dataset/{self.id}/use_privacy_mode", requests.get
+        )[DATASET_USE_PRIVACY_MODE]
+        self._use_privacy_mode = response
+        return self._use_privacy_mode  # type: ignore
+
```

Comment:
I'm a little vary about using round-trip checks for every append. I think this will introduce more flakiness in our pipelines given how often we're seeing 503's etc.

Ideally we'd return this for the dataset constructor  -> then we always have the data locally.
We should the very least cache this locally since this won't ever change.

Reply:
Good point, I changed `client.create_dataset` such that the properties `name`, `is_scene`, and `use_privacy_mode` are saved and dont require round-trip checks